index.html

```diff
@@ -259,10 +259,198 @@ <h1>Architecture</h1>
+	to implement an HTTP client to be able to support Discovery.
+	Another common way to configure a client/consumer is simply to read a (single) TD from a file.
+	There are also other valid ways to do discovery over CoAP, for example using CoRE-RD to discover
+	multiple TD links.
+	This is why in the following we allow any protocol to be used, even though
+	our Directory exploration mechanisms require HTTP.
+	</p>
+	<p class="ednote" title="D-Consumer">
+	Currently the WoT Architecture specification does not clearly say that Consumers MUST support Discovery.  
+	This is the reason for the introduction of the D-Client and D-Consumer terms.
+	</p>
+	</p>
+	We will use the term <a>D-Consumer</a> to refer to a WoT Consumer supporting WoT Discovery,
+	and D-Client as a more general term relating to any client supporting WoT Discovery.
+	Every D-Consumer is also a D-Client.  The rest of this section refers only to D-Clients but all
+	assertions also apply to D-Consumers.
+	<p>
+	As described in section <a href="#architecture"></a>, WoT Discovery has two phases, Introduction
+	and Exploration. In the following sections we will give details on various Introduction and Exploration
+	mechanisms. 
+	However, <em>all</em> Introduction mechanisms result one or more URLs which can be used to fetch TDs by
+	a D-Client. 
+	Some of these TDs, however, may link to
+	TDs stored elsewhere (Thing Links) or describe web services that manage sets of TDs (Thing Description Directories, or TDDs).
+	The following assertions describe the responsibilities of a D-Client:
+	</p>
+	<ul>
+		<li><span class="rfc2119-assertion" id="consumer-must-support-intros">
+			A D-Client MUST support at least one Introduction mechanism.</span>
+			The simplest Introduction mechanism, Direct, simply provides a single URL of a target TD.  
+			This assertion results in different minimal requirements depending on
+			which Introduction mechanism is selected out of the several available.
+			For example, when Direct is used as the sole Introduction mechanism,
+			at a minimum a D-Client must be able to accept a single URL pointing at a TD.
+			Note, however, that this TD could still be describing a Directory.
+			Another option would be for a D-Client to implement, as its sole 
+			Introduction mechanism, an "automatic" Introduction mechanism such as 
+			DNS-SD, possibly in combination with mDNS. 
+			In this case the application 
+			code running in the D-Client
+			would not have to supply any additional information to invoke Discovery.
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-must-support-fetching">
+			A D-Client MUST support fetching a TD from least one URL provided as part of the Introduction process.</span>
+			In other words, not only must a D-Client accept URLs pointing at TDs, it must be
+			able to fetch that TD, for example by using a GET for an HTTP URL or by reading it from a file 
+			specified by way of a file URL.
```

Comment:
```suggestion
			specified by way of a file URI.
```
https://en.wikipedia.org/wiki/File_URI_scheme

Reply:
*Thought* I fixed this, let me checked (won't merge suggesting in case it screws up my other commits... but will take into account in my edits)